circuits/passportVerification/passportVerificationBuilder.circom

```diff
@@ -1,21 +1,126 @@
-        DG1_SIZE,
-        DG15_SIZE,
-        ENCAPSULATED_CONTENT_SIZE,
-        SIGNED_ATTRIBUTES_SIZE,
-        HASH_BLOCK_SIZE,
-        HASH_TYPE,
-        SIGNATURE_TYPE,
-        CHUNK_SIZE,
-        CHUNK_NUMBER,
-        TREE_DEPTH,
-        AA_FLOWS_NUMBER,
-        AA_FLOWS_BITMASK,
-        NoAA_FLOWS_NUMBER,
-        NoAA_FLOWS_BITMASK
+    DG1_SIZE,
+    DG15_SIZE,
+    ENCAPSULATED_CONTENT_SIZE,
+    SIGNED_ATTRIBUTES_SIZE,
+    HASH_BLOCK_SIZE,
+    HASH_TYPE,
+    SIGNATURE_TYPE,
+    SALT_LEN,
+    E_BITS,
+    CHUNK_SIZE,
+    CHUNK_NUMBER,
+    TREE_DEPTH,
+    AA_FLOWS_NUMBER,
+    AA_FLOWS_BITMASK,
+    NoAA_FLOWS_NUMBER,
+    NoAA_FLOWS_BITMASK
 ) {
 
+    var DG1_LEN                  = DG1_SIZE                  * HASH_BLOCK_SIZE;
+    var DG15_LEN                 = DG15_SIZE                 * HASH_BLOCK_SIZE;
+    var ENCAPSULATED_CONTENT_LEN = ENCAPSULATED_CONTENT_SIZE * HASH_BLOCK_SIZE;
+    var SIGNED_ATTRIBUTES_LEN    = SIGNED_ATTRIBUTES_SIZE    * HASH_BLOCK_SIZE;
+
+    var PUBKEY_LEN;
+    var SIGNATURE_LEN;
+
+    //ECDSA
+    if (SIGNATURE_TYPE > 5){
+        PUBKEY_LEN    = 2 * CHUNK_NUMBER * CHUNK_SIZE;
+        SIGNATURE_LEN = 2 * CHUNK_NUMBER * CHUNK_SIZE;   
+    }
+    //RSA
+    if (SIGNATURE_TYPE <= 5){
+        PUBKEY_LEN    = CHUNK_NUMBER;
+        SIGNATURE_LEN = CHUNK_NUMBER;
+    }
+
+
+    signal input encapsulatedContent         [ENCAPSULATED_CONTENT_LEN];
+    signal input dg1                         [DG1_LEN                 ];
+    signal input dg15                        [DG15_LEN                ];
+    signal input signedAttributes            [SIGNED_ATTRIBUTES_LEN   ];
+    signal input signature                   [SIGNATURE_LEN           ];
+    signal input pubkey                      [PUBKEY_LEN              ];
+    // signal input slaveMerkleInclusionBranches[TREE_DEPTH              ];
+    // signal input slaveMerkleRoot;
+
+    signal output passportHash;
+    
+
+    component dg1PassportHasher  = PassportHash(HASH_BLOCK_SIZE, DG1_SIZE,                  HASH_TYPE);
+    component dg15PassportHasher = PassportHash(HASH_BLOCK_SIZE, DG15_SIZE,                 HASH_TYPE);
+    component ecPassportHasher   = PassportHash(HASH_BLOCK_SIZE, ENCAPSULATED_CONTENT_SIZE, HASH_TYPE);
+    component saPassportHasher   = PassportHash(HASH_BLOCK_SIZE, SIGNED_ATTRIBUTES_SIZE,    HASH_TYPE);
+
+    signal                 dg1Hash[HASH_TYPE];
+    signal                dg15Hash[HASH_TYPE];
+    signal encapsulatedContentHash[HASH_TYPE];
+    signal    signedAttributesHash[HASH_TYPE];
+
+    dg1PassportHasher.in   <== dg1;
+    dg1PassportHasher.out  ==> dg1Hash;
+    dg15PassportHasher.in  <== dg15;
+    dg15PassportHasher.out ==> dg15Hash;
+    ecPassportHasher.in    <== encapsulatedContent;
+    ecPassportHasher.out   ==> encapsulatedContentHash;
+    saPassportHasher.in    <== signedAttributes;
+    saPassportHasher.out   ==> signedAttributesHash;
+
+
+    var  FIRST_SHIFT;
```

Comment:
FIRST, SECOND, THIRD.. not informative -> DG1, DG15, ENCAPSULATED_CONTENT

Reply:
Done